app/src/main/java/edu/umd/cmsc436/mstestsuite/MainActivity.java

```diff
@@ -245,6 +248,38 @@ public void onClick(View v) {
     }
 
     @Override
+    public void showHistoryDialog(final String user) {
+        final Dialog dialog = new AppCompatDialog(this);
+        final String [] app_array = getResources().getStringArray(R.array.display_names);
+
+        View root = dialog.getLayoutInflater().inflate(R.layout.history_chooser, null, false);
+
+        TextView title = new TextView(getContext());
```

Comment:
Don't create the textview dynamically, add it above the listview in history_chooser.xml.  Then it doesn't even need to be passed to addHeaderView()